### PR TITLE
hash: add functions for prefetching buckets

### PR DESCRIPTION
**Pull request by mengxiang0811**

### CODE REVIEW

File: lib/librte_hash/rte_cuckoo_hash.c [hunk 1]
```diff
@@ -2659,3 +2659,32 @@ rte_hash_iterate(const struct rte_hash *h, const void **key, void **data, uint32
 	(*next)++;
 	return position - 1;
 }
+
+void
+rte_hash_prefetch_buckets(const struct rte_hash *h, hash_sig_t sig)
```
Comment (AltraMayor) [inline lib/librte_hash/rte_cuckoo_hash.c:2664]: Can we make this function inline? If so make it and its sister.

File: lib/librte_hash/rte_cuckoo_hash.c [hunk 2]
```diff
@@ -2659,3 +2659,32 @@ rte_hash_iterate(const struct rte_hash *h, const void **key, void **data, uint32
...
+{
+	uint16_t short_sig = get_short_sig(sig);
+	uint32_t prim_index = get_prim_bucket_index(h, sig);
+	uint32_t sec_index = get_alt_bucket_index(h, prim_index, short_sig);
+
+	const struct rte_hash_bucket *primary_bkt = &h->buckets[prim_index];
+	const struct rte_hash_bucket *secondary_bkt = &h->buckets[sec_index];
```
Comment (AltraMayor) [inline lib/librte_hash/rte_cuckoo_hash.c]: Drop variables `primary_bkt` and `secondary_bkt` and call `rte_prefetch0()` directly. Do the same for `rte_hash_prefetch_buckets_non_temporal()`.

**Review by AltraMayor — CHANGES_REQUESTED** [review]
**hash: add functions for prefetching buckets**

File: lib/librte_hash/rte_cuckoo_hash.c (hunk 1 above)
Comment (mengxiang0811) [reply lib/librte_hash/rte_cuckoo_hash.c:2664]: No, as this function calls inline functions only available in ``rte_cuckoo_hash.c``.

**Comment by mengxiang0811** [comment]
This rebase is ready for another review.

**Comment by AltraMayor** [comment]
This pull request can be merged.

**Pull request merged**